library/lib.mli

```diff
@@ -87,31 +80,53 @@ val is_modtype : unit -> bool
 val is_modtype_strict : unit -> bool
 val is_module : unit -> bool
 
-(** Returns the opening node of a given name *)
-val find_opening_node : Id.t -> node
+(** The [StagedLibS] abstraction describes operations and traversal for Lib at a
+    given stage. *)
+module type StagedLibS = sig
```

Comment:
Perhaps `type classified_objects` could be moved into this module type. Then the 2 separate instances of it would be nonconvertible and it would be clear in the type of `end_compilation` which part of the tuple is which stage.

Reply:
Thanks, looking much better now:
```ocaml
val end_compilation : DirPath.t -> Nametab.object_prefix * Interp.classified_objects * Synterp.classified_objects
```